crates/router/src/connector/stax/transformers.rs

```diff
@@ -23,6 +23,7 @@ pub struct StaxPaymentsRequest {
     is_refundable: bool,
     pre_auth: bool,
     meta: StaxPaymentsRequestMetaData,
+    reference: String,
```

Comment:
Hi @ManobhavSachan  ,

In reference to issue https://github.com/juspay/hyperswitch/issues/2348 , we need to adjust the reference field mapping based on the API documentation of the connector. Here merchant reference id would be `idempotency_id `

You can find the reference for Stax API documentation [here](https://staxapi.docs.apiary.io/#reference/0/payments/charge-a-payment-method)

Reply:
Thank You for the feedback @srujanchikke, I have made the necessary changes.